[quant][graphmode] Refactor quantization patterns

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #41049 [quant][graphmode] Fp16 quant support - match numerics with eager mode
* **#40894 [quant][graphmode] Refactor quantization patterns**

Summary:

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22403901](https://our.internmc.facebook.com/intern/diff/D22403901)

supriyar committed:
[quant][graphmode] Refactor quantization patterns
Summary:

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Refactor quantization patterns"


Summary:

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -1027,5 +1027,92 @@ graph(%packed_params, %a, %dtype_fp16, %dtype_fp32, %false):
+        return (%r) )";
+
+  return {
+      {"conv1d_prepack_unpack", conv1d_with_quant, conv1d_with_quant_prepack},
```
looks like quantized op is not used actually, I'm also thinking of removing this field and we can make the struct more general to all users of subgraph rewriter

jerryzh168 reviewed (approved):
Thanks

supriyar committed:
Update on "[quant][graphmode] Refactor quantization patterns"


Summary:

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Refactor quantization patterns"


Summary:

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22403901](https://our.internmc.facebook.com/intern/diff/D22403901)

[ghstack-poisoned]
